### ISSUE TITLE
[Bug]: Creating user session with U2F results in cryptic error message

### ISSUE THREAD

**Issue by zapling** [zitadel/zitadel#6784]
### Preflight Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions
- [ ] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

Self-hosted

### Version

2.37.2

### Describe the problem caused by this bug

Trying to login a user with U2F, but I'm getting a cryptic error message and is unable to get the web auth challenge to continue the login process in my custom UI.

### To reproduce

1. Setup U2F (yubikey) on user account  (In my case `andreas.palm+test@fabled.se`)
2. Try and create session
```sh
curl --request POST \
  --url https://<DOMAIN>/v2beta/sessions \
  --header 'Accept: application/json' \
  --header 'Authorization: Bearer <TOKEN>' \
  --header 'Content-Type: application/json' \
  --data '{
  "checks": {
    "user": {
      "loginName": "andreas.palm+test@fabled.se"
    }
  },
  "metadata": {},
  "challenges": {
        "webAuthN": {
            "domain": "localhost",
            "userVerificationRequirement": "USER_VERIFICATION_REQUIREMENT_DISCOURAGED"
        }
    }
}'
```

Error response:
```json
{
    "code":9,
    "message":"Errors.ie4Ai.NotFound (COMMAND-Df4b3)",
    "details":[{"@type":"type.googleapis.com/zitadel.v1.ErrorDetail","id":"COMMAND-Df4b3","message":"Errors.ie4Ai.NotFound"}]
}
```

### Expected behavior

Either that it works or I get a better error message describing what I'm doing wrong :)

### Relevant Configuration

To authorize the API call I'm using a service user that is in a different org than the target user (`andreas.palm+test@fabled.se`), I have configured this service user to have instance level role `IAM User Manager`.

Creating a session on the same user with TOTP works.

### Additional Context

I had a search of the error in the zitadel source code and found [this](https://github.com/zitadel/zitadel/blob/7b91d90eb2f6ba97e519bef6a0fad639a0b42444/internal/command/session.go#L264) that seems to indicate that the user `andreas.palm+test@fabled.se` is inactive, but when I look in Zitadels UI it shows as active, and the API responds `USER_STATE_ACTIVE` as well.

**Comment by hifabienne** [comment]
hei @zapling 
Yes the error message there is wrong. I added a PR to fix that: https://github.com/zitadel/zitadel/pull/6787
The error message just tells you that the user could not be found, this is either the user doesn't exist or is in another state then active.
With what do you try to login? Is it the email address of the user? Or the username?
Authenticating with an email adress does only work if the email address is unique

**Comment by hifabienne** [comment]
@livio-a @muhlemmer Can you help here?

**Comment by livio-a** [comment]
@zapling quick question as we had a similar issue in the past: is the service account (doing the request) part of the same organisation as the user you're trying to login?

**Comment by zapling** [comment]
@livio-a It is not, the service user is indeed in a different org than the user. I have configured the service user to have the instance level role `IAM User Manager`.

**Comment by livio-a** [comment]
> @livio-a It is not, the service user is indeed in a different org than the user. I have configured the service user to have the instance level role `IAM User Manager`.

ok thanks for the quick reply... i'll need to debug (probably tomorrow)

**Comment by zapling** [comment]
Thanks for looking into it, let me know if there is tests I can help with on my end.

**Comment by zapling** [comment]
@livio-a Have you had time to look at this? Do you have a rough idea where one should start looking, if I where to investigate on my end aswell?

**Comment by livio-a** [comment]
Hey @zapling busy and could not work on the issue, but i've now found the problem and started working on it. We should be able to provide a solution by the end of the week. I've just pushed a WIP: https://github.com/zitadel/zitadel/pull/6889

**Comment by livio-a** [comment]
He @zapling the fix was released in 2.41.1 and that was also deployed to zitadel.cloud

**Comment by zapling** [comment]
@livio-a Thank you so much, we will hopefully upgrade our instance version this coming week to confirm.
